### DISCUSSION TITLE
Face recognition , how many images should train for one human?

### DISCUSSION THREAD

**Discussion by neonleo** [emgucv/emgucv#477] [General]
I have working on face recognition attendance system by emgucv, however the result is not accurate as expected.
I try all of 3 algorithm provided by emgucv eigenfaces, fisher, LBHP, with 100 human photo and each of human trained with 3 images. however, none of 3 algorithm achieve over 80% accuracy.

I do some research on codeproject, some documentory of face recognition, and turn out some new technology like facenet, natural network , deep learning is recommended.

I also read some post on stackoverflow said 1 human should be train with 10+ image to get best result, but it not quite realistic when new comer on boarding and take 10 photo to her. 

So is it any success case whether opensource or commercial project using emgucv?

**Comment by emgucv** [comment]
It not just the number of images, the quality of the images are important. The images need to have:
1. Proper alignments (Proper projection to align eye, nose and lips to fixed location on the training and detection images).
2. Facial images from slightly different angles
3. Color and brightness normalization

Read https://docs.opencv.org/4.5.2/da/d60/tutorial_face_main.html
Look at the image sample from the Yale face dataset. If your alignment of facial features are not as good as those images you will get much worst performance.

To align facial feature, consider using DNN face detector and Facial Landmark extraction:
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Models/Dnn/FaceAndLandmarkDetector.cs

Apple's face id (diffent algorightm that also use depth sensor) use a lots of images (at least 10+) during initialization. So it is not un-reasonale to ask user to register faces with more than 10 photos. It is more about how to provide a better user experience during the registration process.

**Reply by neonleo** [reply]
thanks you for answer, I will have a look and try on DNN for face detection and align image.
May I know more about DNN? 
Does DNN help on recognition ( multl scale detect ) by LBPH? Or I can using any method instead of LBPH? Thanks you.
